sql/fixidx/fix_field_indexes.go

```diff
@@ -180,8 +181,17 @@ func FixFieldIndexesForExpressions(logFn func(string, ...any), node sql.Node, sc
 		schemas = append(schemas, child.Schema())
 	}
 
-	if len(schemas) < 1 {
-		return node, transform.SameTree, nil
+	if len(schemas) < 1 && ctx.Version == sql.VersionExperimental {
```

Comment:
Missing the case for len(schemas) < 1 for non-experimental?

Reply:
old way errors if we index childless nodes, but needed shunt for new way. fixing indexing is the long-term solution